code/game/machinery/computer/HolodeckControl.dm

```diff
@@ -453,6 +455,14 @@
 	return 0
 
 /obj/item/weapon/holo/esword/attack(target as mob, mob/user as mob)
+	var/area/myarea = null
+	myarea = get_area(src.loc)
+	if(!istype(myarea, /area/holodeck))
+		force = 0
+		user.Stun(10)
```

Comment:
Yes really. They're intentionally exploiting to get the esword out of the holodeck, walking out with it normally will cause it to delete.

This also has the benefit of ensuring they drop even the referenced invisible item.